dpcpp/matrix/csr_kernels.dp.cpp

```diff
@@ -1625,6 +1625,111 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_CSR_ADVANCED_SPMV_KERNEL);
 
 
+namespace kernel {
+
+
+template <typename IndexType>
+void calc_nnz_in_span(const span row_span, const span col_span,
+                      const IndexType* __restrict__ row_ptrs,
+                      const IndexType* __restrict__ col_idxs,
+                      IndexType* __restrict__ nnz_per_row,
+                      sycl::nd_item<3> item_ct1)
+{
+    const auto tidx = thread::get_thread_id_flat(item_ct1) + row_span.begin;
+    if (tidx < row_span.end) {
+        nnz_per_row[tidx - row_span.begin] = zero<IndexType>();
+        for (size_type col = row_ptrs[tidx]; col < row_ptrs[tidx + 1]; ++col) {
+            if (col_idxs[col] >= col_span.begin &&
+                col_idxs[col] < col_span.end) {
+                nnz_per_row[tidx - row_span.begin]++;
+            }
+        }
+    }
+}
+
+GKO_ENABLE_DEFAULT_HOST(calc_nnz_in_span, calc_nnz_in_span);
+
+
+template <typename ValueType, typename IndexType>
+void compute_submat(size_type num_rows, size_type num_cols, size_type num_nnz,
```

Comment:
Maybe here consistent names for the kernel implementation can be used, cf. with `compute_submatrix_idxs_and_vals` in the cuda version.